### PR TITLE
Adds new configs PoC

### PR DESCRIPTION
**Pull request by jeremy-then**
*rskj:ARROWHEAD-6.3.0-rc*
*ci:630-rc*
*rit:new-configs*

### CODE REVIEW

File: src/main/java/co/rsk/federate/BtcToRskClient.java [hunk 1]
```diff
@@ -206,35 +215,44 @@ public void updateBridge() {
 
         // Call receiveHeaders
         try {
-            int numberOfBlocksSent = updateBridgeBtcBlockchain();
-            logger.debug("[updateBridge] Updated bridge blockchain with {} blocks", numberOfBlocksSent);
+            logger.trace("shouldUpdateBridgeBtcBlockchain: " + shouldUpdateBridgeBtcBlockchain);
+            if(shouldUpdateBridgeBtcBlockchain) {
```
Comment (marcos-iov) [inline src/main/java/co/rsk/federate/BtcToRskClient.java:219]: Not a big deal, but maybe the `if` should go outside the try block

File: src/main/java/co/rsk/federate/BtcToRskClient.java [hunk 2]
```diff
@@ -102,31 +105,37 @@ protected BtcToRskClient(
...
         this.peginInstructionsProvider = peginInstructionsProvider;
         this.federation = federation;
-        this.isUpdateBridgeTimerEnabled = isUpdateBridgeTimerEnabled;
-        this.amountOfHeadersToSend = amountOfHeadersToSend;
+        this.isUpdateBridgeTimerEnabled = config.isUpdateBridgeTimerEnabled();
+        this.amountOfHeadersToSend = config.getAmountOfHeadersToSend();
+        this.shouldUpdateBridgeBtcBlockchain = config.shouldUpdateBridgeBtcBlockchain();
+        this.shouldUpdateBridgeBtcCoinbaseTransactions = config.shouldUpdateBridgeBtcCoinbaseTransactions();
+        this.shouldUpdateBridgeBtcTransactions = config.shouldUpdateBridgeBtcTransactions();
+        this.shouldUpdateCollections = config.shouldUpdateCollections();
```
Comment (marcos-iov) [inline src/main/java/co/rsk/federate/BtcToRskClient.java:113]: Maybe move all this to a method?

File: src/test/java/co/rsk/federate/BtcToRskClientTest.java [hunk 3]
```diff
@@ -2432,6 +2465,40 @@ void updateBridge_when_does_not_hasBetterBlockToSync_updates_headers_coinbase_tr
         verify(federatorSupport, times(1)).sendUpdateCollections();
     }
 
+    @Test
+    void updateBridge_whenUpdateBridgeConfigAreFalse_shouldNotCallAny() throws Exception {
```
Comment (josedahlquist) [inline src/test/java/co/rsk/federate/BtcToRskClientTest.java:2469]: Could you add one more test where the config settings are not defined in the file? According to your change if that's the case the services should run well.

Thanks

**Review by josedahlquist — COMMENTED** [review]
LGTM

File: src/test/java/co/rsk/federate/BtcToRskClientTest.java (hunk 3 above)
Comment (jeremy-then) [reply src/test/java/co/rsk/federate/BtcToRskClientTest.java:2474]: Updated!

File: src/main/java/co/rsk/federate/BtcToRskClient.java (hunk 2 above)
Comment (jeremy-then) [reply src/main/java/co/rsk/federate/BtcToRskClient.java]: Updated!

File: src/main/java/co/rsk/federate/BtcToRskClient.java (hunk 1 above)
Comment (jeremy-then) [reply src/main/java/co/rsk/federate/BtcToRskClient.java]: Updated!

File: src/main/java/co/rsk/federate/BtcToRskClient.java [hunk 4]
```diff
@@ -204,41 +200,50 @@ public void updateBridge() {
...
         logger.debug("[updateBridge] Updating bridge");
 
-        // Call receiveHeaders
-        try {
-            int numberOfBlocksSent = updateBridgeBtcBlockchain();
-            logger.debug("[updateBridge] Updated bridge blockchain with {} blocks", numberOfBlocksSent);
-        } catch (Exception e) {
-            logger.error(e.getMessage(), e);
-            panicProcessor.panic("btclock", e.getMessage());
+        if(shouldUpdateBridgeBtcBlockchain) {
```
Comment (apancorb) [inline src/main/java/co/rsk/federate/BtcToRskClient.java:203]: nit: for the if conditions
```suggestion
        if (shouldUpdateBridgeBtcBlockchain) {
```

File: src/main/java/co/rsk/federate/config/FedNodeSystemProperties.java [hunk 5]
```diff
@@ -97,4 +97,25 @@ public Duration getPegoutSignedCacheTtl() {
         return Duration.ofMinutes(
             getInt("federator.pegoutSignedCacheTtlInMinutes", 30));
     }
+
+    public boolean shouldUpdateBridgeBtcBlockchain() {
```
Comment (apancorb) [inline src/main/java/co/rsk/federate/config/FedNodeSystemProperties.java:101]: The default value are all essentially `true` right?

File: src/test/java/co/rsk/federate/BtcToRskClientTest.java [hunk 6]
```diff
@@ -59,6 +64,7 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import static java.util.Objects.nonNull;
```
Comment (apancorb) [inline src/test/java/co/rsk/federate/BtcToRskClientTest.java:67]: nit: I would probably move this with the rest of the static imports

**Review by apancorb — DISMISSED** [review]
LGTM, left minor comments

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by lucasvuotto** [comment]
pipeline: run

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**